audit/doc.go

```diff
@@ -0,0 +1,204 @@
+// limitations under the License.
+
+/*
+	Package audit implements audit logging.
+
+	This package provides an implementation of the Logger interface that can
+	be used to perform audit logging.
+
+	Entries
+
+	The default logger implementation will create entries in the log file with the following fields.
+
+		time: 	 The time that the action took place.
+		level: 	 The log level for the action.  Warnings will be for failed actions, and info for successful actions.
+		action:	 The action being performed (e.g. add, remove, update, start, stop, etc).
+		type:    Entity being changed if applicable (e.g. resourcepool, host, service, etc).
+		id:      Id of entity being changed if applicable.
+		msg:     User friending message about what action took place.
+		success: Either true or false.
+		user: 	 The user performing the action.  This will default to "system" if no user is provided.
+
+	An entry for a successful attempt to add a resouce pool.
+
+		time="2017-05-11T19:41:10Z" level=info msg="Adding Resource Pool Swimming" action=add success=true user=system type=resourcepool id=Swimming
+
+	An entry for a failed attempt to add a resouce pool.
+
+		time="2017-05-11T19:41:10Z" level=warning msg="Adding Resource Pool Swimming" action=add success=false user=system type=resourcepool id=Swimming
+
+	API
+
+	The Logger interface provides a fluent API for creating log entries.  A new Logger can be retrieved by calling the NewLogger method.
+
+		var auditLogger = audit.NewLogger()
+
+	The default implementation that is returned is a wrapped logri Logger.  Logri is a package owned by Zenoss that adds additional
+	functionality to Loggers from the third party package, logrus.
+
+	Contextual information can be passed into the Logger through the Context method.  The Context method will take a Context interface that is defined in
+	this package.   Currently the context only takes a user.  In the future, additional contextual information can be added to this object.  A new context
+	object can be retrieved through the NewContext method.
+
+		var ctx = audit.NewContext("Alice")
+
+	The context, like the other fields is optional.  If one is not provied the user will default to "system".  To set the Context call the following.
+
+		auditLogger = auditLogger.Context(ctx)
+
+	A friendly message can be set on the audit logger through the use of the "Message" method.  This will set the "msg" field:
+
+		auditLogger = auditLogger.Message("Adding Resource Pool: " + entity.ID)
+
+	The "action field is set through the "Action" method.  Some constants are provided in this package to normalized the values.  Examples are "add", "update",
+	"delete", "start", "stop", etc.  If addition actions need to be added, new constants should be added to this package.
+
+		auditLogger = auditLogger.Action(audit.Add)
+
+	The type of entity being modified is set through the "Type" method.  To normalize these values, constants found in the domain package should be used.
```

Comment:
The discussion should start with the `Entity()` method, and then mention `Type()` and `ID()` as alternatives if the caller can't use `Entity()`